js/src/wasm/WasmDebug.cpp

```diff
@@ -279,6 +279,12 @@ void DebugState::clearBreakpointsIn(JSFreeOp* fop, WasmInstanceObject* instance,
 }
 
 void DebugState::toggleDebugTrap(uint32_t offset, bool enabled) {
+  // Refuse to enable debug traps when recording/replaying. Sometimes these get
+  // enabled for unclear reasons, causing large performance degradations.
+  if (mozilla::recordreplay::IsRecordingOrReplaying()) {
+    enabled = false;
```

Comment:
Is it worth logging something to help us figure out why this is happening?

Reply:
I don't think so, if we want to figure this out we should get STR for the problem, or permission to locally replay one of the affected recordings (though that would be pretty annoying to deal with).  There isn't any indication here about why the debug trap is being enabled, and trying to follow all the callers gets a little bewildering pretty quickly.